File: make_contract/templates/config/entities.ts
export default {
    storage: {
        derive: '#[derive(BorshSerialize, BorshDeserialize, Default, Serialize, Deserialize)]',
        serde: '#[serde(crate = "near_sdk::serde")]',
        struct: 'pub struct $name {$content}',
        initContent: (name: string, dataType: string) => {
            return `pub ${name} ${dataType}, `
        }
    },
    enum: {
        derive: '#[derive(BorshSerialize, BorshDeserialize)]',
        struct: 'pub enum $name {$content}',
        initContent: (numName: string, objType: string) => {
            const obj = !!objType ? `(${objType})` : ''
            return numName + obj + ','
        }
    },
    contract: {
        header: '#[near_bindgen]',
        derive: '#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]',
        struct: 'pub struct $name {$content}',
        initContent: (name: string, dataType: string) => {
            return `${name} ${dataType}, `
        }
    }
}

File: make_contract/templates/counter/src/entities.rs
pub use crate::*;

//{$entities}

//{$impl_entities}

#[derive(BorshDeserialize, BorshSerialize, Default)]
pub struct OldCounter {
    pub value: u8,
    pub new_value: u8
}

#[derive(BorshSerialize, BorshDeserialize, Default, Serialize, Deserialize)]
#[serde(crate = "near_sdk::serde")]
pub struct AccountV1 {
    pub vote: u8,
    pub balance: u8
}

#[derive(BorshSerialize, BorshDeserialize, Default, Serialize, Deserialize)]
#[serde(crate = "near_sdk::serde")]
pub struct AccountV2 {
    pub vote: u8,
    pub balance: u8,
    pub bio: String
}

#[derive(BorshSerialize, BorshDeserialize, Default, Serialize, Deserialize)]
#[serde(crate = "near_sdk::serde")]
pub struct Account {
    pub vote: u8,
    pub balance: u8,
    pub bio: String,
    pub last_change: BlockHeight
}

#[derive(BorshSerialize, BorshDeserialize)]
pub enum VersionedAccount {
    V1(AccountV1),
    V2(AccountV2),
    Current(Account)
}

